### PR TITLE
Add Kitchen CI & Yum repos

### PR DESCRIPTION
**Pull request by epcim**
Added some basic kitchen CI configuration and yum repos + fixes to pass converge on centos 7.1.

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -3,13 +3,22 @@ driver:
...
   - name: ubuntu-12.04
+  - name: ubuntu-14.04
   - name: centos-6.4
+  - name: centos-7.1
 
 suites:
   - name: default
     run_list:
+      - zabbix_ng::default
+      #- zabbix_ng::sender
```
Comment (chr4) [inline .kitchen.yml]: I think it's better to either mark commented out blocks like this with a `TODO` or simply remove them altogether. Similar like this:

``` ruby
# TODO: zabbix_ng::sender
```

File: metadata.rb [hunk 2]
```diff
@@ -4,6 +4,9 @@
 license          'GNU Public License 3.0'
 description      'Installs/Configures zabbix_ng'
 long_description 'Installs/Configures zabbix_ng'
-version          '0.1.1'
+version          '0.1.3'
```
Comment (chr4) [inline metadata.rb]: Remove the version bump altogether, as with this, `0.1.2` will never be in the supermarket. I think it's better to just leave the version bump itself to the maintainer.

File: recipes/repository.rb [hunk 3]
```diff
@@ -19,9 +19,31 @@
 #
 
 # Official Zabbix repository
-apt_repository 'zabbix' do
-  uri 'http://repo.zabbix.com/zabbix/2.4/ubuntu'
-  distribution node['lsb']['codename']
-  components %w(main)
-  key 'http://repo.zabbix.com/zabbix-official-repo.key'
+# TODO, move repo keys and url to attributes
```
Comment (chr4) [inline recipes/repository.rb]: Minor nag: use `TODO: Move...` :)

File: recipes/repository.rb [hunk 4]
```diff
@@ -19,9 +19,31 @@
...
+    distribution node['lsb']['codename']
+    components %w(main)
+    key 'http://repo.zabbix.com/zabbix-official-repo.key'
+  end
+when 'redhat', 'centos', 'fedora'
+  yum_repository 'zabbix' do
+    description "Official zabbix repository"
+    baseurl 'http://repo.zabbix.com/zabbix/2.4/rhel/7/x86_64'
+    gpgkey 'http://repo.zabbix.com/zabbix-official-repo.key'
+    action :create
```
Comment (chr4) [inline recipes/repository.rb]: Out of interest: is `action :create` required for `yum_repository`? As with `apt_repository` this is the default. If yum defaults to `:create` as well, we should either use the action nowhere, or in all repository resources.

File: recipes/server.rb [hunk 5]
```diff
@@ -20,8 +20,14 @@
 
 include_recipe 'zabbix_ng::repository'
 
+#package 'zabbix-server-mysql'
```
Comment (chr4) [inline recipes/server.rb]: Just remove this. I don't see an advantage of using mysql, if we one day offer support for it, we should rather use an attribute that specifies the server package to install.

**Comment by chr4** [comment]
Hi, thanks for your contribution!
I basically think this is good work and a good idea, I added some minor thoughts in the code comments.
Please review and add your thoughts!

**Comment by chr4** [comment]
What does the kitchen CI actually test? We didn't implement any tests, did we?
If it doesn't actually check anything besides that the recipe runs through, we should split the kitchen CI into a seperate pull request and merge it later when there's actual integration/ unit tests available.

**Comment by epcim** [comment]
Thanks for the comments. Well mostly the things I forgot in code while already working on another branch/feature. I suppose they are now mostly fixed, except TODO, vs TODO: (anyway some removed in future PR).

Kitchen verifier just now does nothing, except test for successful converge. In my wrapper I have now an recipe for simple mysql/(pgsql) database. I will consider to share these, but better into kitchen "test" cookbook & integration testing, than as a recipe in zabbix_ng (as everybody has different preferences regarding DB configuration).

Wait a day or so before mege - as I still running "kitchen test". Anyway I will provide soon another PR for "rhel" platform family and a PR with zabbix version specified in attributes. These stacks in this order.

**Comment by epcim** [comment]
well the tests passed on my side, the current conflict is due updates in your master now ( regarding zabbix_version attr ). 

BTW: can you pls share your idea about versioning, aka PR #3 is OK or the way it's done violate your personal flavors ;-) ?

**Comment by chr4** [comment]
Sorry, I'm a little confused by the many pull-requests. #3 seems to be the latest one? Or did you split them up? Can you close the ones that are not up to date please?
I'll comment on #3, as I think this is the current one. If I'm wrong, please advise :)

**Comment by epcim** [comment]
#3 is the latest, it's based on #2 and that on #1 (it would be too complex to split them as they interact). Sorry for confusion. My original aim was not to do much, but I ended up with three major updates as I was implementing it localy.

**Pull request closed without merging**